### ISSUE TITLE
Remote with UNC path not working

### ISSUE THREAD

**Issue by steelwil** [git-for-windows/git#505]
If I look in the config file I correctly see something like: 

    \\\\192.168.227.205\\myproject

git remote -v is also correct

    $ git remote -v
    MyuncRemote \\192.168.227.205\myproject(fetch)
    MyuncRemote \\192.168.227.205\myproject(push)

but when I do a fetch I get the following error:

    $ git fetch MyuncRemote
    fatal: 'MyuncRemote' does not appear to be a git repository
    fatal: Could not read from remote repository.

    Please make sure you have the correct access rights
    and the repository exists.

when I try to clone I get the following:

    $ git clone \\\\\192.168.227.205\\myproject
    Cloning into 'myproject'...
    fatal: '\192.168.227.205\myproject' does not appear to be a git repository
    fatal: Could not read from remote repository.

if I try: 

    $ git clone \\192.168.227.205\myproject
    fatal: repository '\192.168.227.205myproject' does not exist

if I add 4 extra \ to the remote name in the config file I am able to fetch:

    	url = \\\\\\\\192.168.227.205\\myproject

I could not get clone to work using any combination.

I also only have "read" access to the path, but I only ever want to fetch/pull.

**Comment by nalla** [comment]
Have you tried `url = //192.168.227.205/myproject` ?

**Comment by steelwil** [comment]
Hi forward slashes work perfectly but I am unsure it is still a UNC name.

**Comment by nalla** [comment]
well at least for `git clone` http://www.mingw.org/wiki/Posix_path_conversion clearly states that you should use forward slashes for UNC paths.

**Comment by dscho** [comment]
I could imagine that this is at least related to https://github.com/git-for-windows/git/issues/439

**Comment by besmirzanaj** [comment]
so this is not an issue or a bug, right?

**Comment by steelwil** [comment]
I feel that it can be handled better, as most windows users will make this mistake. An error message or an automatic conversion would be good.

**Comment by dscho** [comment]
@steelwil I agree, but please note that you are asking for *a lot*. I spent some 2h yesterday to investigate and had to stop there because I had to take care of other, more pressing things (you have a work-around, and you could have taken the opportunity to participate by hunting the bug yourself, those are the main reasons why this ticket appears not all that pressing to me). The point where I left off showed that `git-upload-pack '\\127.0.0.1\test'` is spawned via `mingw_spawn` using *the shell*, therefore MSys2's runtime enters the game and it seems that it is responsible for stripping one of the backslashes. From the strace output:

```
...
   92  729189 [main] sh 5140 build_argv: cmd = 'sh -c "git-upload-pack '\\127.0.0.1\test'" "git-upload-pack '\\127.0.0.1\git'"', winshell = 1, glob = 1
  125  729314 [main] sh 5140 build_argv: argv[0] = 'sh'
   65  729379 [main] sh 5140 build_argv: argv[1] = '-c'
  133  729512 [main] sh 5140 lstat64: entering
   68  729580 [main] sh 5140 normalize_posix_path: src git-upload-pack '\127.0.0.1\test'
...
```

**Comment by steelwil** [comment]
Hi dscho 

I appreciate the time you have spent with this. I will use the work around. Sure I could of participated in bug hunting myself and I would loved to have rather submitted a patch. 

I would like to see the product improve so if a bug is at least submitted everybody will know of the issue. Maybe something as simple as updating a FAQ or release notes would do.

Regards

**Comment by dscho** [comment]
> I would like to see the product improve so if a bug is at least submitted everybody will know of the issue. Maybe something as simple as updating a FAQ or release notes would do.

Excellent. Looking forward to reading it.

**Comment by wilson0x4d** [comment]
Neither format appears to work anymore, using: git version 2.10.2.windows.1

Path such as `//MYSAN/PATH/Project.git` (ie. `file:////MYSAN/PATH/project.git`) works fine in TortoiseGit.

Remotes don't show anything fancy/abnormal.

Can fetch, push, pull from Tortoise but not command-line.

**Comment by dscho** [comment]
@wilson0x4d There was a UNC-related breakage that was fixed in v2.11.0(2). But that is already quite old, there have been several releases since. The current Git for Windows version is v2.12.2(2).

You need to test whether your problems persist with the current version, and if they do, open a new ticket with verbose information. In particular, you definitely want to specify *how* it "does not work". Pasting the output of a failing command is probably the best.
